Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -45,100 +46,83 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
+                    authorizationSegment.SubjectEndpoint as AuthorizationSegmentEndpointWithValue;
 
-                    string edOrgTypes = string.Join(
-                        "', '",
-                        authorizationSegment.ClaimsEndpoints
-                            .Select(s => s.Name.TrimSuffix("Id"))
-                            .Distinct()
-                            .OrderBy(x => x));
+                // This should never happen
+                if (subjectEndpointWithValue == null)
+                {
+                    throw new Exception(
+                        "The claims-based authorization segment subject endpoint for a single-item authorization was not defined with a value.");
+                }
 
+                if (subjectEndpointWithValue.Value == null)
+                {
                     throw new EdFiSecurityException(
-                        $"Unable to authorize the request because there is no authorization support for associating the "
-                        + $"API client's associated education organization types ('{edOrgTypes}') with the resource.");
+                        $"Access to the resource item could not be authorized because the '{subjectEndpointWithValue.Name}' of the resource is empty.");
                 }
 
-                // Combine multiple statements resulting from multiple EdOrg types in the API client's claims using "OR"
-                // (This forces at least 1 of the relationships with the EdOrg types to exist)
-                segmentStatements.Add(string.Join($"{Environment.NewLine}OR ", segmentExpressions));
+                // Perform defensive checks against the remote possibility of SQL injection attack
+                ValidateTableNameParts(claimEndpointName, subjectEndpointName, authorizationSegment.AuthorizationPathModifier);
+
+                string derivedAuthorizationViewName = ViewNameHelper.GetFullyQualifiedAuthorizationViewName(
+                    subjectEndpointName,
+                    claimEndpointName,
+                    authorizationSegment.AuthorizationPathModifier);
+
+                if (!IsAuthorizationViewSupported(derivedAuthorizationViewName))
+                {
+                    unsupportedAuthorizationViews.Add(derivedAuthorizationViewName);
+
+                    edOrgTypes.AddRange(authorizationSegment.ClaimsEndpoints
+                       .Select(s => s.Name.TrimSuffix("Id")));                    
+
+                    continue;
+                }
+
+                segmentStatements.Add(CreateSegmentExpression(ref parameterIndex));
+
+                string CreateSegmentExpression(ref int index)
+                {
+                    return string.Format(
+                        StatementTemplate,
+                        GetMultiValueCriteriaExpression(authorizationSegment.ClaimsEndpoints.ToList(), parameters, ref index),
+                        subjectEndpointName.ContainsIgnoreCase("EducationOrganizationId")
```

Comment:
I'm asking questions here... not telling you to change anything yet.

Is `ContainsIgnoreCase` the right function here? True question. First, why is it "Contains"? Are you expecting to be handling a value with a prefix or suffix applied? Next, shouldn't this logic apply to any be any EdOrgId (or EdOrg-derived Id)? If the subject endpoint is an EdOrg (e.g. SchoolId), then shouldn't the value be supplied to the template format as "TargetEducationOrganizationId" too? This doesn't seem cohesive, but I'm only looking at this in GitHub right now.

One other thought: if this logic changes, let's try to avoid checking the subject endpoint multiple times to see if it is an EdOrg. We've already done it above and perhaps you save the results of that into a `bool` parameter and use that instead of performing multiple checks. This particular code is going to be running _a lot_.

Reply:
It should be EqaulIgnoreCase instead of ContainsIgnoreCase. when subjectEndpointName is "EducationOrganizationId" and then we are calling tuple table and column name is TargetEducationOrganizationId  otherwise any person type like StaffUSI or ParentUSI or StudentUSI . So that's the reason for this change